### PR TITLE
Add tests for rules with sudouser with uid, gid and netgroup

### PR DESCRIPTION
**Pull request by jakub-vavra-cz**
## Summary by Sourcery

Extend sudo integration test coverage for sudoUser, sudoRunAsUser, sudoRunAsGroup, and hostname handling, and remove a redundant non-POSIX group sudo test.

Enhancements:
- Adjust hostname exclusion sudo rule tests to use combined ALL,!host syntax to reflect supported LDAP backend behavior.

Tests:
- Add tests for sudo rules targeting users by UID and groups by GID in sudoUser, including non-POSIX groups and netgroups.
- Add tests for excluded users and groups in sudoUser when using ALL with negations.
- Add tests for sudoRunAsUser and sudoRunAsGroup entries specified by numeric UID/GID.
- Remove the older standalone non-POSIX group sudoUser test in favor of the new basic sudo coverage.

### CODE REVIEW

**Review by shridhargadekar — CHANGES_REQUESTED** [review]
Please check the comments

File: pytest/tests/test_basic.py [hunk 1]
```diff
@@ -139,6 +171,83 @@ def test_basic__single_group(client: Client, provider: GenericProvider):
...
+        1. Run "sudo /bin/ls /root" as user-1 (member of non-POSIX group-1)
+        2. Run "sudo /bin/ls /root" as user-deny (not in group)
+    :expectedresults:
+        1. User in non-POSIX group is able to run /bin/ls as root
+        2. User not in the group is not able to run /bin/ls as root
+    :customerscenario: False
+    """
+    _setup_sudo(client, provider)
+    u = provider.user("user-1").add(uid=10001, gid=10001)
+    u_deny = provider.user("user-deny").add(uid=10002, gid=10002)
```
Comment (shridhargadekar) [inline pytest/tests/test_basic.py]: these POSIX attributes become redundant with `ldap_id_mapping = false`

File: pytest/tests/test_basic.py [hunk 2]
```diff
@@ -139,6 +171,83 @@ def test_basic__single_group(client: Client, provider: GenericProvider):
...
+    g = provider.group("group-1").add().add_member(u)
+    provider.sudorule("test").add(user=g, host="ALL", command="/bin/ls")
+    client.sssd.domain["ldap_id_mapping"] = "false"
+    client.sssd.restart()
+
+    if isinstance(provider, AD):
```
Comment (shridhargadekar) [inline pytest/tests/test_basic.py]: If it's decided to execute this against `AD`, then `if` condition can be dropped. and execute `id` commands as it is

Comment (jakub-vavra-cz) [reply pytest/tests/test_basic.py]: done

File: pytest/tests/test_basic.py (hunk 1 above)
Comment (jakub-vavra-cz) [reply pytest/tests/test_basic.py]: They are needed when ldap_id_mapping = false and redundant when ldap_id_mapping = true .

File: pytest/tests/test_basic.py [hunk 3]
```diff
@@ -58,6 +58,38 @@ def test_basic__single_user(client: Client, provider: GenericProvider):
     ), f"User {u2.name} was able to run sudo with command /bin/ls!"
 
 
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.BareLDAP)
+@pytest.mark.topology(KnownTopology.BareClient)
+def test_basic__single_user_by_uid(client: Client, provider: GenericProvider):
+    """
+    :title: Sudo rule allows user specified by UID in sudoUser
```
[the remark this answers is not in the document]
Comment (shridhargadekar) [reply pytest/tests/test_basic.py]: In ideal scenario, all local users should be unique. However it's a corner case, where duplicate users have same name with different UIDs.
Current TC looks appropriate as well.

**Review by shridhargadekar — CHANGES_REQUESTED** [review]

**Review by shridhargadekar — CHANGES_REQUESTED** [review]

**Review by shridhargadekar — CHANGES_REQUESTED** [review]

File: pytest/tests/test_basic.py [hunk 4]
```diff
@@ -383,84 +641,127 @@ def test_basic__multiple_runasuser(client: Client, provider: GenericProvider):
...
 
+    res = client.auth.sudo.run_advanced(u1.name, "Secret123", parameters=["-g", g1.name], command="id -g")
+    assert res.rc == 0, f"User {u1.name} failed to run sudo with command id -g as {g1.name}!"
+    assert "20001" in res.stdout.strip(), f"id -g mismatch: expected 20001, got {res.stdout!r}"
+
     assert (
-        client.auth.sudo.run_advanced(u1.name, "Secret123", parameters=["-g", g1.name], command="whoami").rc == 0
-    ), f"User {u1.name} failed to run sudo with command whoami as {g1.name}!"
+        not client.auth.sudo.run_advanced(u1.name, "Secret123", parameters=["-g", g2.name], command="id -g").rc == 0
+    ), f"User {u1.name} was able to run sudo with command id -g as {g2.name}!"
```
Comment (shridhargadekar) [inline pytest/tests/test_basic.py]: Negative message missing

**Review by shridhargadekar — CHANGES_REQUESTED** [review]

**Review by shridhargadekar — CHANGES_REQUESTED** [review]

**Review by shridhargadekar — APPROVED** [review]
